addons/sale_coupon/models/sale_order.py

```diff
@@ -405,14 +414,19 @@ def update_line(order, lines, values):
                             value_found = True
                             # Working on Case 3.
                             lines_to_remove -= line
+                            lines_to_keep += [(0, False, value)]
```

Comment:
This whole change (with below lines) is "just" for perfs right? Move it in a separate commit please.